sklearnex/spmd/decomposition/tests/test_incremental_pca_spmd.py

```diff
@@ -0,0 +1,266 @@
+    assert_allclose(incpca.explained_variance_, incpca_spmd.explained_variance_)
+    assert_allclose(
+        incpca.explained_variance_ratio_, incpca_spmd.explained_variance_ratio_
+    )
+
+
+@pytest.mark.skipif(
+    not _mpi_libs_and_gpu_available,
+    reason="GPU device and MPI libs required for test",
+)
+@pytest.mark.parametrize(
+    "dataframe,queue",
+    get_dataframes_and_queues(dataframe_filter_="dpnp,dpctl", device_filter_="gpu"),
+)
+@pytest.mark.parametrize("whiten", [True, False])
+@pytest.mark.parametrize("n_components", [None, 2, 5])
+@pytest.mark.parametrize("num_samples", [100, 200])
+@pytest.mark.parametrize("num_features", [10, 20])
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+@pytest.mark.mpi
+def test_incremental_pca_fit_spmd_random(
+    dataframe, queue, whiten, n_components, num_samples, num_features, dtype
+):
+    # Import spmd and non-SPMD algo
+    from sklearnex.preview.decomposition import IncrementalPCA
+    from sklearnex.spmd.decomposition import IncrementalPCA as IncrementalPCA_SPMD
+
+    tol = 7e-5 if dtype == np.float32 else 1e-7
+
+    # Create data and process into dpt
+    X = _generate_statistic_data(num_samples, num_features, dtype)
+    dpt_X = _convert_to_dataframe(X, sycl_queue=queue, target_df=dataframe)
+    X_test = _generate_statistic_data(num_samples // 5, num_features, dtype)
+    dpt_X_test = _convert_to_dataframe(X_test, sycl_queue=queue, target_df=dataframe)
+    local_X = _get_local_tensor(X)
+    local_dpt_X = _convert_to_dataframe(local_X, sycl_queue=queue, target_df=dataframe)
+
+    incpca_spmd = IncrementalPCA_SPMD(n_components=n_components, whiten=whiten)
+    incpca = IncrementalPCA(n_components=n_components, whiten=whiten)
+
+    incpca_spmd.fit(local_dpt_X)
+    incpca.fit(dpt_X)
+
+    assert_allclose(incpca.n_components_, incpca_spmd.n_components_, atol=tol)
+    assert_allclose(incpca.components_, incpca_spmd.components_, atol=tol)
+    assert_allclose(incpca.singular_values_, incpca_spmd.singular_values_, atol=tol)
+    assert_allclose(incpca.mean_, incpca_spmd.mean_, atol=tol)
+    assert_allclose(incpca.var_, incpca_spmd.var_, atol=tol)
+    assert_allclose(incpca.explained_variance_, incpca_spmd.explained_variance_, atol=tol)
+    assert_allclose(
+        incpca.explained_variance_ratio_, incpca_spmd.explained_variance_ratio_, atol=tol
+    )
+
+    y_trans_spmd = incpca_spmd.transform(dpt_X_test)
+    y_trans = incpca.transform(dpt_X_test)
+
+    assert_allclose(_as_numpy(y_trans_spmd), _as_numpy(y_trans), atol=tol)
+
+
+@pytest.mark.skipif(
+    not _mpi_libs_and_gpu_available,
+    reason="GPU device and MPI libs required for test",
+)
+@pytest.mark.parametrize(
+    "dataframe,queue",
+    get_dataframes_and_queues(dataframe_filter_="dpnp,dpctl", device_filter_="gpu"),
+)
+@pytest.mark.parametrize("whiten", [True, False])
+@pytest.mark.parametrize("n_components", [None, 2, 5])
+@pytest.mark.parametrize("num_blocks", [1, 2])
```

Comment:
Similar to the comment I make in the IncrementalLinearRegression implementation, would it be worth having a num_blocks==1 test against sklearnex PCA?

Reply:
why is it better then the current version?

Reply:
PCA has been more extensively tested and mathematically validated (3+ years, sklearn conformance, etc)

Reply:
https://github.com/intel/scikit-learn-intelex/blob/main/sklearnex/decomposition/tests/test_pca.py
https://github.com/intel/scikit-learn-intelex/blob/main/sklearnex/preview/decomposition/tests/test_incremental_pca.py
I'd say that IncrementalPCA is more extensively tested

Reply:
Thanks for linking to files, but you haven't proven anything. Please answer the original question

Reply:
what do you mean by original question? your argument was that PCA is more tested than IncrementalPCA and the links I provided shows it looks wrong.

Reply:
Original question:

Similar to the comment I make in the IncrementalLinearRegression implementation, would it be worth having a num_blocks==1 test against sklearnex PCA?

Your response was a question:

why is it better than the current version?

Based on your last comment, I assume the answer to be 'no'. @olegkkruglov

Reply:
as for me, it doesn't matter, both variants are acceptable. so I asked do you have strong arguments for this change. if you don't, then, yes, the answer to the original question is 'no'